test/crawler_web/controllers/keyword_controller_test.exs

```diff
@@ -83,4 +83,70 @@ defmodule CrawlerWeb.KeywordControllerTest do
                gettext("One or more keywords are invalid! Allowed keyword length is 1-100")
     end
   end
+
+  describe "GET show/2" do
+    test "given a keyword belongs to a valid user, renders details page", %{conn: conn} do
+      user = insert(:user)
+
+      keyword =
+        insert(:keyword,
+          user_id: user.id,
+          status: :completed,
+          html: "html",
+          ad_total: 5,
+          urls_ad_top: [],
+          non_ad_count: 10,
+          urls_non_ad: [],
+          ad_top_count: 10,
+          total: 50
+        )
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> get(Routes.keyword_path(conn, :show, keyword))
+
+      response = html_response(conn, 200)
+
+      assert response =~ keyword.name
+      assert response =~ "#{gettext("Top Ad Count:")} #{keyword.ad_top_count}"
+      assert response =~ "#{gettext("Total Ad Count:")} #{keyword.ad_total}"
+      assert response =~ "#{gettext("Non-Ad Count:")} #{keyword.non_ad_count}"
+      assert response =~ "#{gettext("Total:")} #{keyword.total}"
+    end
+
+    test "given keyword with new status, renders details page with searching status", %{conn: conn} do
+      user = insert(:user)
+
+      keyword =
+        insert(:keyword,
+          user_id: user.id,
+          status: :new
+        )
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> get(Routes.keyword_path(conn, :show, keyword))
+
+      response = html_response(conn, 200)
+
+      assert response =~ keyword.name
+      assert response =~ gettext("Searching")
+    end
+  end
+
+  test "given another user keyword, renders home page with an error", %{conn: conn} do
+    another_user = insert(:user)
+    keyword = insert(:keyword, user_id: another_user.id)
+    expected_user = insert(:user)
+
+    conn = conn |> log_in_user(expected_user) |> get(Routes.keyword_path(conn, :show, keyword))
+    assert redirected_to(conn) == Routes.home_path(conn, :index)
+
+    conn2 = get(conn, Routes.home_path(conn, :index))
+
+    assert response = html_response(conn2, 200)
```

Comment:
the best practice is return 404 status if requested resource is not available

Reply:
@Nihisil @andyduong1920 Fixed in [83bca5b](https://github.com/vnntsu/elixir-ic/pull/43/commits/83bca5bcc57439ba6bba96945aa06a8922586c56)